File: solutions/594.longest-harmonious-subsequence/longest-harmonious-subsequence.cpp
class Solution {
public:
    int findLHS(vector<int>& nums) {
        unordered_map<int, int> m;
        for(int num : nums) m[num]++;
        int res = 0;
        for(auto it : m) 
            if(m.count(it.first - 1) > 0) 
                res = max(res, it.second + m[it.first - 1]);
        return res;
    }
};

File: solutions/594.longest-harmonious-subsequence/longest-harmonious-subsequence.java
class Solution {
    public int findLHS(int[] nums) {
        Map<Integer, Integer> map = new HashMap<>();
        for(int num : nums) map.put(num, map.getOrDefault(num, 0) + 1);
        int res = 0;
        for(int num : nums) 
            if(map.containsKey(num + 1)) 
                res = Math.max(res, map.get(num) + map.get(num + 1));
        return res;
    }
}